pgxn/neon/walproposer.c

```diff
@@ -856,46 +863,91 @@ RecvVoteResponse(Safekeeper *sk)
 	 * we are not elected yet and thus need the vote.
 	 */
 	if ((!sk->voteResponse.voteGiven) &&
-		(sk->voteResponse.term > wp->propTerm || wp->n_votes < wp->quorum))
+		(sk->voteResponse.term > wp->propTerm || wp->state == WPS_CAMPAIGN))
```

Comment:
This line is quite hard to think about. The logic here is that we accept rejected votes only after election has finished.

That means if sk will give a vote, then disconnect, then reconnect and reject a vote, the walproposer will throw FATAL and restart the voting. No issue here, just trying to remember the logic.

Reply:
`wp->n_votes < wp->quorum` ---> `wp->state == WPS_CAMPAIGN`

This looks like a logic change to me, but maybe I don't have enough context to quickly grasp it.

@arssher , can you please update the comment?

"or we are not elected yet and thus need the vote"
 This was helpful and matched with an old code, but hard to understand with new one that uses state.

Reply:
Comment is still relevant, Arthur explained correctly. WPS_CAMPAIGN means we are not elected yet (campaigning for election).